python/GangaPanda/Lib/ProdTrans/ProdTransPandaRTHandler.py

```diff
@@ -180,17 +180,23 @@ def prepare(self, app, appsubconfig, appmasterconfig, jobmasterconfig):
                 randomized_lfn = lfn + ('.%s.%d.%s' % (job.backend.site, int(time.time()), commands.getoutput('uuidgen 2> /dev/null')[:4] ) )
             else:
                 randomized_lfn = lfn
+
             ofspec.lfn = randomized_lfn
             randomized_lfns.append(randomized_lfn)
             ofspec.destinationDBlock = jspec.destinationDBlock
             ofspec.destinationSE = jspec.destinationSE
             ofspec.dataset = jspec.destinationDBlock
             ofspec.type = 'output'
             jspec.addFile(ofspec)
+
+            # remove the first section of the file name if it matches the file type
+            if len(randomized_lfn.split('.')) > 0 and randomized_lfn.split('.')[0].find( lfntype ) != -1:
+                randomized_lfn = '.'.join( randomized_lfn.split('.')[1:] )
```

Comment:
Calling `split` on a string will always create a list of at least one element:
```python
>>> 'test'.split('.')
['test']
>>> ''.split('.')
['']
```
so `len(randomized_lfn.split('.')) > 0` will always pass so this could be simplified to:
```python
if lfntype in randomized_lfn.split('.')[0]:
```
However, if `randomized_lfn` has no `.` in it then the call to `randomized_lfn.split('.')[1:]` could fail.